app/controllers/tasks_controller.rb

```diff
@@ -4,7 +4,31 @@ class TasksController < ApplicationController
   before_action :authenticate_user!
 
   def index
-    @tasks = current_user.tasks
-    render json: @user
+    # @task = current_user.tasks
+    @pagy, @tasks = pagy(Task.includes(:creator, :assignee, :parent_task, :subtasks).all)
```

Comment:
extract this to a private method for readability and modularity. 

`Task.includes(:creator, :assignee, :parent_task, :subtasks).all`